core/src/components/button/button.ionic.scss

```
@@ -59,37 +34,60 @@
 
     z-index: 1;
   }
+}
 
-  ::slotted(ion-icon[slot="start"]) {
-    @include globals.margin-horizontal(null, globals.$ionic-space-200);
-  }
+// Button Fills
+// -------------------------------------------------------------------------------
 
-  ::slotted(ion-icon[slot="end"]) {
-    @include globals.margin-horizontal(globals.$ionic-space-200, null);
-  }
+// Solid Button
+// --------------------------------------------------
+
+:host(.button-solid) {
+  --background-activated: #{globals.ion-color(primary, shade)};
+  --background-hover: #{globals.ion-color(primary, contrast)};
+  --background-focused: #{globals.ion-color(primary, contrast)};
+  --background-focused-opacity: 0.24;
+  --background-hover-opacity: 0.08;
+  --ripple-opacity: var(--background-activated-opacity, 1);
+  --ripple-color: var(--background-activated);
+}
+
+// Outline Button
+// --------------------------------------------------
+
+:host(.button-outline) {
+  --border-width: #{globals.$ionic-border-size-025};
+  --border-style: #{globals.$ionic-border-style-solid};
+  --background-activated: #{globals.$ionic-color-neutral-200};
+  --background-focused: transparent;
+  --background-hover: #{globals.ion-color(primary, base)};
+  --background-hover-opacity: 0.04;
+  --ripple-opacity: var(--background-activated-opacity, 1);
+  --ripple-color: var(--background-activated);
 }
 
-// Button Colors
+// Clear Button
 // --------------------------------------------------
 
-// Solid Button with Color
-:host(.button-solid.ion-color) .button-native {
-  background: globals.current-color(base);
-  color: globals.current-color(contrast);
+:host(.button-clear) {
+  --background-activated: #{globals.$ionic-color-neutral-200};
+  --background-focused: transparent;
+  --background-hover: #{globals.ion-color(primary, base)};
+  --background-hover-opacity: 0.04;
+  --ripple-opacity: var(--background-activated-opacity, 1);
+  --ripple-color: var(--background-activated);
 }
 
-// Outline Button with Color
-:host(.button-outline.ion-color) .button-native {
-  border-color: globals.current-color(base);
+// Ripple Effect
+// -------------------------------------------------------------------------------
 
-  background: transparent;
-  color: globals.current-color(base);
+:host(.button-solid.ion-color) ion-ripple-effect {
+  color: #{globals.current-color(shade)};
```

```suggestion
  color: globals.current-color(shade);
```

Escapes are only required when used with css variables